### DISCUSSION TITLE
POD is not created always when a change is committed in GIt

### DISCUSSION THREAD

**Discussion by aupadh12** [argoproj/argo-cd#5710] [Q&A]
Hi Team,

We have noticed that argocd does not always create a new pod after we make changes in the manifests files. As a result, the required changes are not sent to the live state.
It only takes effect when we delete the pod and a new pod is created.
We would like this to happen automatically.
Can someone please look into it?

Regards,
Ashutosh Upadhyay

**Comment by jkleinlercher** [comment]
Which changes do you mean? Changes in the Deployment/Pod spec should restart the pod automatically. Changes in config maps, secrets, ingress etc. do not restart pods automatically and I do think that works as designed. You do not always want a restart because some applications reload configurations automatically without restarts. However, there are other methods how you can enforce pod restarts yourselves. 
e.g. https://sanderknape.com/2019/03/kubernetes-helm-configmaps-changes-deployments/

**Reply by peternann** [reply]
"You do not always want a restart..." But sometimes you do, and you know you do!

I think this would be a great option to add to Argo CD. This has already bitten a project I am on twice - Updating a configmap and believing Argo CD will do the right ting for us to make it stick, but it does not.

It's not really appropriate to refer to 3rd-party tools - This is precisely the sort of thing you expect/need Argo CD to make simple (and safe) for you, when you take it on.

An option per app to 'restart on changes to related resources' or something like that would be a godsend I reckon!

**Comment by aupadh12** [comment]
By chanages I mean the version of the images. For example, I have image version passed into the deployment as a volume mount where a file called application.yml is created. The data in this application.yml is coming from the configmap. So, when I change the image version in my configmap, the pods which are already running needs to be restarted in order for the new changes to take place. Do you think argo rollout can help here ?

**Reply by jannfis** [reply]
I think the pointers given by @jkleinlercher are quite good to get started. If you do not use Helm, there is some third party tooling out there, the most well known one probably being https://github.com/stakater/Reloader. This tool watches for changes to ConfigMaps and then can restart some workload in your cluster according to its configuration.

**Reply by peternann** [reply]
As I mentioned above, this is something I really expected Argo CD to do, and am quite surprised it does not. 3rd-party tooling? - Bleh! That's what we are trying to avoid with Argo CD!
